napari/_qt/qt_dims_slider.py

```diff
@@ -0,0 +1,471 @@
+        return self._fps
+
+    @fps.setter
+    def fps(self, value):
+        self._fps = value
+        self.play_button.fpsspin.setValue(abs(value))
+        self.play_button.reverse_check.setChecked(value < 0)
+        self.fps_changed.emit(value)
+
+    @property
+    def loop_mode(self):
+        return self._loop_mode
+
+    @loop_mode.setter
+    def loop_mode(self, value):
+        self._loop_mode = value
+        self.play_button.mode_combo.setCurrentIndex(value)
+        self.mode_changed.emit(value)
+
+    @property
+    def frame_range(self):
+        frame_range = (self._minframe, self._maxframe)
+        frame_range = frame_range if any(frame_range) else None
+        return frame_range
+
+    @frame_range.setter
+    def frame_range(self, value):
+        if not isinstance(value, (tuple, list, type(None))):
+            raise TypeError('frame_range value must be a list or tuple')
+        if value and not len(value) == 2:
+            raise ValueError('frame_range must have a length of 2')
+        if value is None:
+            value = (None, None)
+        self._minframe, self._maxframe = value
+        self.range_changed.emit(tuple(value))
+
+    def _update_play_settings(self, fps, loop_mode, frame_range):
+        if fps is not None:
+            self.fps = fps
+        if loop_mode is not None:
+            self.loop_mode = loop_mode
+        if frame_range is not None:
+            self.frame_range = frame_range
+
+    def _play(
+        self,
+        fps: Optional[float] = None,
+        loop_mode: Optional[str] = None,
+        frame_range: Optional[Tuple[int, int]] = None,
+    ):
+        """Animate (play) axis. Same API as QtDims.play()
+
+        Putting the AnimationWorker logic here makes it easier to call
+        QtDims.play(axis), or hit the keybinding, and have each axis remember
+        it's own settings (fps, mode, etc...).
+        """
+
+        # having this here makes sure that using the QtDims.play() API
+        # keeps the play preferences synchronized with the play_button.popup
+        self._update_play_settings(fps, loop_mode, frame_range)
+
+        # setting fps to 0 just stops the animation
+        if fps == 0:
+            return
+
+        worker, thread = new_worker_qthread(
+            AnimationWorker,
+            self,
+            start=True,
+            connections={'frame_requested': self.qt_dims._set_frame},
+        )
+        worker.finished.connect(self.qt_dims.stop)
+        thread.finished.connect(self.play_stopped.emit)
+        self.play_started.emit()
+        self.thread = thread
+        return worker, thread
+
+
+class QtCustomDoubleSpinBox(QDoubleSpinBox):
+    """ Custom Spinbox that emits an editingFinished event both on the regular
+    editing event, as well as on the valueChanged event IF the left mouse
+    button is down.  On the standard class, the valueChanged event is too
+    spammy for the lineEdit, but the editingFinished does not pay attention
```

Comment:
I have to say I don't quite get this - is it for the frame rate setting? are you getting too many events that things crash / slow down?

Reply:
sorry that wasn't clear. The behavior I wanted was for it _not_ to trigger on valueChanged when typing, because each keypress is going to cause a dramatic visual change (for instance, the "1" in 10 is going to cause a 1 second pause before it resumes when you hit the "0"... typing something like "0.5" is also problematic).  So I went for editingFinished... however, then the spinbox buttons don't work.  Turns out there are no signals for "spinBox.clicked"... Similarly, if you just use the mousePress event, then you can't hold down the mouse on the spinBox and increase the FPS.  It's all minor stuff, but this was the best I came up with to achieve the behavior I wanted.  If the extra code is not worth that slight behavioral "nicety", we can go back to valueChanged.

Reply:
ah that makes more sense, because this can be edited during playing which might trigger all sorts of unpleasant behavior. this is good, maybe make that more clear in the comment and then it's good to stay, it's a nice touch